src/types/access.ts

```diff
@@ -56,3 +57,44 @@ export interface InternalAccess extends Access {
 }
 
 export type AppAccess = PublicAccess | RestrictedAccess | InternalAccess;
+
+export const AppAccess = {
+  /**
+   * Validate the CIDR rules for AppAccess.
+   *
+   * Public - always valid. No CIDR rules are needed here, and the type-system enforces this
+   * Restricted - Must not include global CIDR ranges as doing so makes other listed CIDR ranges redundant
+   * Internal - All CIDR ranges must start `10.`
+   *
+   * @throws `Error` when CIDR rules are invalid
+   */
+  validate: (access: AppAccess) => {
+    switch (access.scope) {
+      case AccessScope.RESTRICTED: {
+        const ranges = access.cidrRanges.map((_) => _.uniqueId);
+        const rangesString = ranges.join(", ");
+
+        if (ranges.includes(Peer.anyIpv4().uniqueId) || ranges.includes(Peer.anyIpv6().uniqueId)) {
```

Comment:
I think this PR is also adding IPv6 support to this check? 💯

Reply:
Oh. Yes. I guess this is _not_ a no-op after all then! Happy to remove IPv6 stuff or update the PR as `fix:` to create a release.

Reply:
I think it's fine to include it here, I don't think many people will be specifying IPv6 ranges yet! Feel free to update to `fix` if you want to generate a release though!

Reply:
A new release for this is most likely to just be noise. Will leave as is.